Amplify/Core/Support/AmplifyTask+OperationTaskAdapters.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+class AmplifyOperationTaskAdapter<Request: AmplifyOperationRequest, Success, Failure: AmplifyError>: AmplifyTask {
+    let operation: AmplifyOperation<Request, Success, Failure>
+    let childTask: ChildTask<Void, Success, Failure>
+    var resultToken: UnsubscribeToken? = nil
+
+    init(operation: AmplifyOperation<Request, Success, Failure>) {
```

Comment:
```suggestion
    public init(operation: AmplifyOperation<Request, Success, Failure>) {
```

Reply:
the constructor needs to be public as well for the plugins to use

Reply:
The instance is only created internally so the init should not have to be public.

Reply:
In the plugin code, for example, https://github.com/aws-amplify/amplify-ios/pull/2140/files#diff-fa2f4789a9a71f094ae985b1d3b57e4c8b52caef6e7cf6fea0c6234652e803e9

```
func query<R: Decodable>(request: GraphQLRequest<R>) async throws -> GraphQLTask<R>.Success {
        let operation = AWSGraphQLOperation(request: request.toOperationRequest(operationType: .query),
                                            session: session,
                                            mapper: mapper,
                                            pluginConfig: pluginConfig,
                                            resultListener: nil)
        let task = AmplifyOperationTaskAdapter(operation: operation)
        queue.addOperation(operation)
        return try await task.result
    }
```

how should the plugins be creating the task after creating the operation?